x/concentrated-liquidity/lp_test.go

```diff
@@ -551,6 +551,95 @@ func (s *KeeperTestSuite) TestIsInitialPosition() {
 	}
 }
 
+func (s *KeeperTestSuite) TestUpdatePosition() {
+	type updatePositionTest struct {
+		poolId          uint64
+		owner           sdk.AccAddress
+		lowerTick       int64
+		upperTick       int64
+		liquidityDelta  sdk.Dec
+		amount0Expected sdk.Int
+		amount1Expected sdk.Int
+		expectedError   bool
+	}
+
+	tests := map[string]updatePositionTest{
+		"happy path": {
+			poolId:          1,
+			owner:           s.TestAccs[0],
+			lowerTick:       DefaultLowerTick,
+			upperTick:       DefaultUpperTick,
+			liquidityDelta:  DefaultLiquidityAmt,
+			amount0Expected: DefaultAmt0Expected,
+			amount1Expected: DefaultAmt1Expected,
+			expectedError:   false,
+		},
+		"error: invalid pool id": {
+			poolId:          2,
+			owner:           s.TestAccs[0],
+			lowerTick:       DefaultLowerTick,
+			upperTick:       DefaultUpperTick,
+			liquidityDelta:  DefaultLiquidityAmt,
+			amount0Expected: DefaultAmt0Expected,
+			amount1Expected: DefaultAmt1Expected,
+			expectedError:   true,
+		},
+		"new position when calling update position": {
+			poolId:          1,
+			owner:           s.TestAccs[1],
+			lowerTick:       DefaultLowerTick,
+			upperTick:       DefaultUpperTick,
+			liquidityDelta:  DefaultLiquidityAmt,
+			amount0Expected: DefaultAmt0Expected,
+			amount1Expected: DefaultAmt1Expected,
+			expectedError:   false,
+		},
+	}
+	for name, tc := range tests {
+		s.Run(name, func() {
+			tc := tc
+			s.SetupTest()
+
+			// create a CL pool
+			originalPool := s.PrepareConcentratedPool()
+
+			// create position
+			// Fund test account and create the desired position
+			s.FundAcc(s.TestAccs[0], sdk.NewCoins(sdk.NewCoin(ETH, DefaultAmt0), sdk.NewCoin(USDC, DefaultAmt1)))
+			_, _, _, err := s.App.ConcentratedLiquidityKeeper.CreatePosition(s.Ctx, originalPool.GetId(), s.TestAccs[0], DefaultAmt0, DefaultAmt1, sdk.ZeroInt(), sdk.ZeroInt(), tc.lowerTick, tc.upperTick)
+			s.Require().NoError(err)
+
+			// system under test
+			actualAmount0, actualAmount1, err := s.App.ConcentratedLiquidityKeeper.UpdatePosition(
+				s.Ctx,
+				tc.poolId,
+				tc.owner,
+				tc.lowerTick,
+				tc.upperTick,
+				tc.liquidityDelta,
+			)
+
+			if tc.expectedError {
+				s.Require().Error(err)
```

Comment:
should we check what actualAmount0 and actualAmount1 is when we return the error?

Reply:
Exactly, we want to check that they return empty sdk.Ints and not a value

Reply:
gotcha, added